File: 019_Remove_Nth_Node_From_End_of_List/019_1.java
class ListNode 
    {
        int val;
        ListNode next;
        ListNode(int x) { val = x; }
    }

public class test
{
    public static ListNode removeNthFromEnd(ListNode head, int n)
    {
        ListNode ans = new ListNode(0);
        ans.next = head;

        ListNode val1 = ans, val2 =ans;
        int i = 0;
        while (i < n){
            val1 =  val1.next;
            i++;
        }

        while (val1.next != null){
            val1 = val1.next;
            val2 = val2.next;
        }
        val2.next = val2.next.next;
        return ans.next;   
    }

    public static void main(String[] args)
    {
        ListNode a = new ListNode(0);
        ListNode b = new ListNode(1);
        a.next = b;
        ListNode c = new ListNode(3);
        b.next = c;
        ListNode d = new ListNode(4);
        c.next = d;
        ListNode e = new ListNode(5);
        d.next = e;

        a = removeNthFromEnd(a, 2);
        while (a != null)
        {
            System.out.println(a.val);
            a = a.next;    
        }
    }
}

File: 019_Remove_Nth_Node_From_End_of_List/019_1.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def removeNthFromEnd(self, head, n):
        ans = ListNode(0)
        ans.next = head
        val1, val2 = ans, ans

        for _ in range(n):
            val1 = val1.next

        while val1.next:
            val1 = val1.next
            val2 = val2.next

        val2.next = val2.next.next
        return ans.next